src/layers/pool.jl

```diff
@@ -143,3 +143,24 @@ function topk_index(y::AbstractVector, k::Int)
 end
 
 topk_index(y::Adjoint, k::Int) = topk_index(y', k)
+
+"""
+    WeigthAndSumPool(in_feats)
+
+WeigthAndSum sum pooling layer. Compute the weights for each node and perform a weighted sum.
+"""
+struct WeigthAndSumPool
+    in_feats::Int
+end
+
+function (ws::WeigthAndSumPool)(g::GNNGraph, x::AbstractArray)
+    atom_weighting = Dense(ws.in_feats, 1, sigmoid; bias = true)
```

Comment:
the dense layer should be part of the type and created at construction time, not during the forward pass